dist/gcc-compatible/Hacl_Streaming_HMAC.c

```diff
@@ -0,0 +1,2204 @@
+    Hacl_Hash_SHA3_update_last_sha3(Spec_Hash_Definitions_SHA3_512, p1, last, last_len);
+    return;
+  }
+  if (scrut.tag == Blake2S_s)
+  {
+    uint32_t *p1 = scrut.case_Blake2S_s;
+    uint32_t wv[16U] = { 0U };
+    Hacl_Hash_Blake2s_update_last(last_len, wv, p1, false, prev_len, last_len, last);
+    return;
+  }
+  if (scrut.tag == Blake2S_128_s)
+  {
+    Lib_IntVector_Intrinsics_vec128 *p1 = scrut.case_Blake2S_128_s;
+    #if HACL_CAN_COMPILE_VEC128
+    KRML_PRE_ALIGN(16) Lib_IntVector_Intrinsics_vec128 wv[4U] KRML_POST_ALIGN(16) = { 0U };
+    Hacl_Hash_Blake2s_Simd128_update_last(last_len, wv, p1, false, prev_len, last_len, last);
+    return;
+    #else
+    KRML_MAYBE_UNUSED_VAR(p1);
+    return;
+    #endif
+  }
+  if (scrut.tag == Blake2B_s)
+  {
+    uint64_t *p1 = scrut.case_Blake2B_s;
+    uint64_t wv[16U] = { 0U };
+    Hacl_Hash_Blake2b_update_last(last_len,
+      wv,
+      p1,
+      false,
+      FStar_UInt128_uint64_to_uint128(prev_len),
+      last_len,
+      last);
+    return;
+  }
+  if (scrut.tag == Blake2B_256_s)
+  {
+    Lib_IntVector_Intrinsics_vec256 *p1 = scrut.case_Blake2B_256_s;
+    #if HACL_CAN_COMPILE_VEC256
+    KRML_PRE_ALIGN(32) Lib_IntVector_Intrinsics_vec256 wv[4U] KRML_POST_ALIGN(32) = { 0U };
+    Hacl_Hash_Blake2b_Simd256_update_last(last_len,
+      wv,
+      p1,
+      false,
+      FStar_UInt128_uint64_to_uint128(prev_len),
+      last_len,
+      last);
+    return;
+    #else
+    KRML_MAYBE_UNUSED_VAR(p1);
+    return;
+    #endif
+  }
+  KRML_HOST_EPRINTF("KaRaMeL abort at %s:%d\n%s\n",
+    __FILE__,
+    __LINE__,
+    "unreachable (pattern matches are exhaustive in F*)");
+  KRML_HOST_EXIT(255U);
+}
+
+static void finish(Hacl_Agile_Hash_state_s *s, uint8_t *dst)
+{
+  Hacl_Agile_Hash_state_s scrut = *s;
+  if (scrut.tag == MD5_s)
+  {
+    uint32_t *p1 = scrut.case_MD5_s;
+    Hacl_Hash_MD5_finish(p1, dst);
+    return;
+  }
+  if (scrut.tag == SHA1_s)
+  {
+    uint32_t *p1 = scrut.case_SHA1_s;
+    Hacl_Hash_SHA1_finish(p1, dst);
+    return;
+  }
+  if (scrut.tag == SHA2_224_s)
+  {
+    uint32_t *p1 = scrut.case_SHA2_224_s;
+    Hacl_Hash_SHA2_sha224_finish(p1, dst);
+    return;
+  }
+  if (scrut.tag == SHA2_256_s)
+  {
+    uint32_t *p1 = scrut.case_SHA2_256_s;
+    Hacl_Hash_SHA2_sha256_finish(p1, dst);
+    return;
+  }
+  if (scrut.tag == SHA2_384_s)
+  {
+    uint64_t *p1 = scrut.case_SHA2_384_s;
+    Hacl_Hash_SHA2_sha384_finish(p1, dst);
+    return;
+  }
+  if (scrut.tag == SHA2_512_s)
+  {
+    uint64_t *p1 = scrut.case_SHA2_512_s;
+    Hacl_Hash_SHA2_sha512_finish(p1, dst);
+    return;
+  }
+  if (scrut.tag == SHA3_224_s)
+  {
+    uint64_t *p1 = scrut.case_SHA3_224_s;
+    uint32_t remOut = 28U;
+    uint8_t hbuf[256U] = { 0U };
+    uint64_t ws[32U] = { 0U };
+    memcpy(ws, p1, 25U * sizeof (uint64_t));
+    for (uint32_t i = 0U; i < 32U; i++)
+    {
+      store64_le(hbuf + i * 8U, ws[i]);
+    }
+    memcpy(dst + 28U - remOut, hbuf, remOut * sizeof (uint8_t));
```

Comment:
Ditto about the lack of `Hacl_Hash_SHA3_sha512_finish` for this one.

Reply:
Worth fixing, good catch!